### ISSUE TITLE
Thank you!

### ISSUE THREAD

**Issue by CV8R** [Yetangitu/zmapi#1]
Just checking in to say thanks - the zmapi interface is genius and saved me tone of work scripting triggers.

Only thing I did was add -u"username:password" before each curl statement due to the zoneminder server using basic auth.

Cheers!
CV8R

**Comment by Yetangitu** [comment]
Why, thanks for the thanks, much appreciated!

As to using basic auth instead of JWT I can add this to the config file as an option. It is better to have this in a file than on the command line to keep the username/password from being visible to anyone who cares to run `ps auwx` on that box.

**Comment by CV8R** [comment]
Agree in a file. The box actually uses both Basic Auth and ZMs auth.

**Comment by Yetangitu** [comment]
I just pushed some updates which should enable this feature by adding two new parameters to the `zmapi.conf`  config file:

```
proxy_username=george
proxy_password=0rw3ll
```
When these are configured _curl_ will use these (through a ephemeral _netrc_ file) for authentication. This type of authentication can be used in combination with JWT authentication (just configure `username`, `password`, `proxy_username` and `proxy_password`) or on its own (leave `username` and `password` blank). I tested the combination of JWT and basic auth so I can verify this works as intended, I did not test using only basic auth since I do not have a ZM installation which does not require authentication. I see no reason for it not to work though...

Give it a try and tell me about any problems you encounter and/or suggestions you have.
